### ISSUE TITLE
Views: Filter results by matching data with current content

### ISSUE THREAD

**Issue by BWPanda** [backdrop/backdrop-issues#3683]
**Description of the need**
This is something I don't believe is possible with Views (in Backdrop or Drupal), but that I think would be handy. If you want to create a 'related' block, you can filter results based on matching terms in the current node being viewed (e.g. https://drupal.stackexchange.com/a/205924/16537), but as far as I know you can't do this with other data (e.g. content type).

If I want to have all content on my site show a related block next to it, and have that block show other content of the same type, I have to create separate views' displays for each content type. But if this feature were to be added, I could create just one views' display that would work everywhere.

And ideally the same could be done for other data (e.g. content created in the same month as current node, content scheduled to be published on the same day as current node, content with the same field value as current node, etc.).

**Proposed solution**
Basically a views' filter that lets you compare to the same data on the node being viewed (based on the URL I guess).
